{
	"schema_version": "1.6.0",
	"id": "BIT-vault-2024-6468",
	"modified": "2024-07-17T08:13:46.654170Z",
	"published": "2024-07-17T07:39:35.824Z",
	"aliases": [
		"CVE-2024-6468",
		"GHSA-2qmw-pvf7-4mw6",
		"GO-2024-2982"
	],
	"details": "Vault and Vault Enterprise did not properly handle requests originating from unauthorized IP addresses when the TCP listener option, proxy_protocol_behavior, was set to deny_unauthorized. When receiving a request from a source IP address that was not listed in proxy_protocol_authorized_addrs, the Vault API server would shut down and no longer respond to any HTTP requests, potentially resulting in denial of service.While this bug also affected versions of Vault up to 1.17.1 and 1.16.5, a separate regression in those release series did not allow Vault operators to configure the deny_unauthorized option, thus not allowing the conditions for the denial of service to occur.Fixed in Vault and Vault Enterprise 1.17.2, 1.16.6, and 1.15.12.",
	"affected": [
		{
			"package": {
				"ecosystem": "Bitnami",
				"name": "vault",
				"purl": "pkg:bitnami/vault"
			},
			"severity": [
				{
					"type": "CVSS_V3",
					"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
				}
			],
			"ranges": [
				{
					"type": "SEMVER",
					"events": [
						{
							"introduced": "1.10.0"
						},
						{
							"fixed": "1.15.12"
						},
						{
							"introduced": "1.16.0-rc1"
						},
						{
							"fixed": "1.16.3"
						},
						{
							"introduced": "1.17.0-rc1"
						},
						{
							"fixed": "1.17.2"
						}
					]
				}
			],
			"database_specific": {
				"source": "https://github.com/bitnami/vulndb/tree/main/data/vault/BIT-vault-2024-6468.json"
			}
		}
	],
	"references": [
		{
			"type": "WEB",
			"url": "https://discuss.hashicorp.com/t/hcsec-2024-14-vault-vulnerable-to-denial-of-service-when-setting-a-proxy-protocol-behavior/68518"
		}
	],
	"database_specific": {
		"cpes": [
			"cpe:2.3:a:hashicorp:vault:*:*:*:*:*:go:*:*"
		],
		"severity": "High"
	}
}
